src/hooks/useModal/useModal.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright 2024 Mia srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { useCallback, useState } from 'react'
+
+import { ModalAPI } from './useModal.types'
+
+/**
+ * A hook that allows the use of Modal components.
+ *
+ * @returns {ModalAPI} An object which includes the state of the modal (visible or hidden) and methods
+ * for opening, closing, and changing the state of the modal.
+ */
+export const useModal = (): ModalAPI => {
```

Comment:
The current implementation of the useModal we have in our internal repositories is actually a little bugous, since it requires to have both the modal and the hook instantiated and requires the developer to correctly map the modal visible state with the desired modal.

For example I'm always required to something like


```jsx
import { Modal, useModal } from ...

function MyComp({}) {
   const { isModalVisible, open, close, toggle } = useModal()

   return <Modal visible={isModalVisible} />
}
```

It would be nice to have some kind of interface that helps managing such visibility state without the need for the developer remapping.

What if the hook returns the Modal component as well?

So I'd do something like:

```jsx
import { useModal } from ...

function MyComp({}) {
   const { ModalComponent, open, close, toggle } = useModal()

   return <ModalComponent />
}
```